server/src/utils/mixerConnections/LawoRubyConnection.ts

```diff
@@ -392,16 +411,19 @@ export class LawoRubyMixerConnection {
             }
 
             logger.debug(`Subscription of AMix state: ${command}`)
-            this.emberConnection.subscribe(node, () => {
-                logger.trace(`Receiving AMix state from Ch ${ch}`)
-                store.dispatch(
-                    storeSetAMix(
-                        ch - 1,
-                        (node.contents as Model.Parameter).value === 1
+            this.emberConnection.subscribe(
+                node as NumberedTreeNode<EmberElement>,
+                () => {
+                    logger.trace(`Receiving AMix state from Ch ${ch}`)
+                    store.dispatch(
+                        storeSetAMix(
+                            ch - 1,
+                            (node.contents as Model.Parameter).value === true

```

Comment:
Did the type of `(node.contents as Model.Parameter).value` suddenly change??? Everywhere else it's being cast to a `number`?
That seems extremely fishy

Reply:
It did not change, the type can be any of `number | string | boolean | Buffer | null` depending on the remote ember+ tree. We cast to the type we expect based on the knowledge we (as the developers) have of the remote tree.

In hindsight we should have probably typed the ember+ library a bit different in order to do cheap type assertions in library consumers. Although that really wouldn't have saved me from the mistake I made here, it just means we wouldn't have to use as many type casts.